src/renderer/containers/Account/AccountOverview/index.tsx

```diff
@@ -5,56 +5,101 @@ import axios from 'axios';
 
 import DetailPanel from '@renderer/components/DetailPanel';
 import Qr from '@renderer/components/Qr';
-import {getActivePrimaryValidatorConfig} from '@renderer/selectors';
+import {useBooleanState} from '@renderer/hooks';
+import {getActivePrimaryValidatorConfig, getManagedAccounts} from '@renderer/selectors';
 import {formatAddress} from '@renderer/utils/address';
 
 import './AccountOverview.scss';
 
 const AccountOverview: FC = () => {
   const {accountNumber} = useParams();
-  const activePrimaryValidator = useSelector(getActivePrimaryValidatorConfig);
   const [balance, setBalance] = useState<number | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [signingKeyVisible, toggleSigningKeyVisible] = useBooleanState(false);
+  const activePrimaryValidator = useSelector(getActivePrimaryValidatorConfig);
+  const managedAccounts = useSelector(getManagedAccounts);
+  const managedAccount = managedAccounts[accountNumber];
+
+  useEffect(() => {
+    if (signingKeyVisible) toggleSigningKeyVisible();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [accountNumber]);
 
   useEffect(() => {
     if (!activePrimaryValidator) return;
 
     const fetchData = async (): Promise<void> => {
       const {ip_address: ipAddress, port, protocol} = activePrimaryValidator;
       const address = formatAddress(ipAddress, port, protocol);
+
+      setLoading(true);
       const {data} = await axios.get(`${address}/account_balance/${accountNumber}`);
       setBalance(data.balance);
+      setLoading(false);
     };
 
     fetchData();
   }, [accountNumber, activePrimaryValidator]);
 
+  const getItems = () => {
+    let items = [
+      {
+        key: 'Balance',
+        value: loading ? '-' : balance || '0',
```

Comment:
For this `'-'` and the other one below, a few notes:
- it's tricky to decide whether to just re-render or provide loading state when they navigate between accounts since the request is so fast everything just flickers
- allowing a plain re-render makes the software just seem kind of laggy (since it updates in ~0.2 seconds)
- loading state of `null` causes heights to change/elements to bounce do to missing strings
- loading state of `<Loader />` looks like something is blinking, not visible long enough to comprehend what it is